lib/repo-type-mappings/yocto-based-OS-image/versionist.conf.js

```diff
@@ -56,8 +61,14 @@ module.exports = {
   },
   incrementVersion: (currentVersion, incrementLevel) => {
     if (isESR(currentVersion)) {
-      return semver.inc(currentVersion, 'patch')
+      if (isValidSemver(currentVersion) != null) {
+        return semver.inc(currentVersion, 'patch')
+      } else {
+        const patchVersion = Number(currentVersion.split('.')[2])
+        return currentVersion.slice(0, currentVersion.length - 1) + (patchVersion + 1)
+      }
```

Comment:
Instead of this we could modify `balena-semver.inc()` to increment legacy ESR versions without making them semver compatible.